src/test/components/ActiveTabTimeline.test.js

```diff
@@ -148,5 +152,80 @@ describe('ActiveTabTimeline', function() {
       fireEvent.click(getGlobalTrackRow());
       expect(getSelectedThreadIndex(getState())).toBe(threadIndex);
     });
+
+    it('does not display the resources panel if there are no resource tracks', () => {
+      const { getState, container } = setup();
+      expect(getActiveTabResourceTracks(getState()).length).toBe(0);
+      expect(container.querySelector('.timelineResourcesHeader')).toBeFalsy();
+    });
+  });
+
+  describe('ActiveTabResourcesPanel', function() {
+    function setup() {
+      const { profile, ...pageInfo } = addActiveTabInformationToProfile(
+        getProfileWithNiceTracks()
+      );
+      // Setting the first thread as parent track and the second as the iframe track.
+      profile.threads[0].frameTable.innerWindowID[0] =
+        pageInfo.parentInnerWindowIDsWithChildren;
+      profile.threads[1].frameTable.innerWindowID[0] =
+        pageInfo.iframeInnerWindowIDsWithChild;
+      const store = storeWithProfile(profile);
+      store.dispatch(
+        changeTimelineTrackOrganization({
+          type: 'active-tab',
+          browsingContextID: pageInfo.firstTabBrowsingContextID,
+        })
+      );
+      const { getState, dispatch } = store;
+      const resourceTracks = getActiveTabResourceTracks(getState());
+
+      const renderResult = render(
+        <Provider store={store}>
+          <ActiveTabResourcesPanel
+            resourceTracks={resourceTracks}
+            setIsInitialSelectedPane={() => {}}
+          />
+        </Provider>
+      );
+
+      const { container } = renderResult;
+      const getResourcesPanelHeader = () =>
+        ensureExists(
+          container.querySelector('.timelineResourcesHeader'),
+          `Couldn't find the resources panel header with selector .timelineResourcesHeader`
+        );
+
+      return {
+        ...renderResult,
+        ...pageInfo,
+        dispatch,
+        getState,
+        profile,
+        store,
+        getResourcesPanelHeader,
+      };
+    }
+
+    it('matches the snapshot of a resources panel', () => {
+      const { container } = setup();
+      expect(container.firstChild).toMatchSnapshot();
+    });
+
+    it('is closed by default', () => {
+      const { getResourcesPanelHeader } = setup();
+      expect(getResourcesPanelHeader().classList.contains('opened')).toBe(
+        false
+      );
+    });
+
+    it('clicking on the header opens the resources panel', () => {
+      const { getResourcesPanelHeader } = setup();
+      const resourcesPanelHeader = getResourcesPanelHeader();
+      expect(resourcesPanelHeader.classList.contains('opened')).toBe(false);
```

Comment:
TODO - Once there is actual content here, it would be better to use user-like actions, such as checking the existence of something, rather than checking for a class name. This will make the test more robust to implementation detail changes.

Reply:
You're right. Since there is nothing else to check currently, I had to use the class, but we should fix this once we have more stuff in there. Added a TODO for this.